telegram/_telegramobject.py

```diff
@@ -105,9 +132,39 @@ def _apply_api_kwargs(self) -> None:
           then you can pass everything as proper argument.
         """
         # we convert to list to ensure that the list doesn't change length while we loop
-        for key in list(self.api_kwargs.keys()):
+        for key in list(api_kwargs.keys()):
             if getattr(self, key, True) is None:
-                setattr(self, key, self.api_kwargs.pop(key))
+                setattr(self, key, api_kwargs.pop(key))
+
+    def __setattr__(self, key: str, value: object) -> None:
+        """ "Overrides :meth:`object.__setattr__` to prevent the overriding of attributes.
+
+        Raises:
+            :exc:`AttributeError`
+        """
+        # protected attributes can always be set for convenient internal use
+        if (key == "_frozen") or (not getattr(self, "_frozen", True)) or key.startswith("_"):
```

Comment:
```suggestion
        if key[0] == "_" or not getattr(self, "_frozen", True):
```
a slight simplification (notice reorder and indexing). 
Also using indexing instead of startswith is about 2x faster on py3.11:
``` python
>>> timeit.timeit("a.startswith('_')", setup="a='_key'")
0.049614202000014984
>>> timeit.timeit("a[0] == '_'", setup="a='_key'")
0.02290340099997934
```
same can be done for `__delattr__`